s3_bucket: don't always call policy/requestPayment/versioning API

pilou- opened the pull request:
##### SUMMARY
don't call `policy`/`requestPayment`/`versioning` API when default value is used with parameters `policy`/`requester_pays`/`versioning`.

Fix the following error:
```
The full traceback is:
Traceback (most recent call last):
  File "/tmp/ansible_s3_bucket_payload_tfn8z6cy/__main__.py", line 181, in create_or_update_bucket
    requester_pays_status = get_bucket_request_payment(s3_client, name)
  File "/tmp/ansible_s3_bucket_payload_tfn8z6cy/ansible_s3_bucket_payload.zip/ansible/module_utils/cloud.py", line 150, in retry_func
    raise e
  File "/tmp/ansible_s3_bucket_payload_tfn8z6cy/ansible_s3_bucket_payload.zip/ansible/module_utils/cloud.py", line 140, in retry_func
    return f(*args, **kwargs)
  File "/tmp/ansible_s3_bucket_payload_tfn8z6cy/__main__.py", line 314, in get_bucket_request_payment
    return s3_client.get_bucket_request_payment(Bucket=bucket_name).get('Payer')
  File "$HOME/.venv/ansible/lib/python3.6/site-packages/botocore/client.py", line 310, in _api_call
    return self._make_api_call(operation_name, kwargs)
  File "$HOME/.venv/ansible/lib/python3.6/site-packages/botocore/client.py", line 599, in _make_api_call
    raise error_class(parsed_response, operation_name)
botocore.exceptions.ClientError: An error occurred (NotImplemented) when calling the GetBucketRequestPayment operation: The requested resource is not implemented
```

##### ISSUE TYPE
- Bugfix Pull Request

##### COMPONENT NAME
s3_bucket

##### ANSIBLE VERSION
<!--- Paste verbatim output from "ansible --version" between quotes -->
```paste below
2.8
```

##### ADDITIONAL INFORMATION
With #46745, it allows to use `s3_bucket` with [Scaleway Object Storage](https://www.scaleway.com/object-storage/) (if you're willing to test these changes with the Scaleway service, there is currently a free public beta).

sieben commented:
I get the following message:

```
fatal: [localhost]: FAILED! => {"changed": false, "msg": "Couldn't connect to AWS: Invalid endpoint: s3.nl-ams.scw.cloud"}
```

pilou- commented:
@sieben you need to test both this PR and #46745. The error you encountered is fixed by #46745.

sieben commented:
https://github.com/ansible/ansible/pull/46353/files

mattclay commented:
CI failure in integration tests: https://app.shippable.com/github/ansible/ansible/runs/88071/67/tests

pilou- committed:
s3_bucket: handle not implemented operations

Don't fail when policy, requestPayment, tagging or versioning API is not
implemented by the endpoint and if related parameters policy, requester_pays,
tags or versioning are None.

pilou- commented:
Rebased in order to trigger shippable.

mattclay commented:
Closing and re-opening to trigger CI.
